Sample not working

Hi,

I think this sample is not working. I tried in my machine how says readme file, but without successful.

The log can be find here: 

```
$ mvn package test appengine:devserver
```

```
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building guestbook 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ guestbook ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ guestbook ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ guestbook ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:testCompile (default-testCompile) @ guestbook ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ guestbook ---
[INFO] Surefire report directory: /Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/target/surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.google.appengine.demos.guestbook.GuestbookServletTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.413 sec
Running com.google.appengine.demos.guestbook.SignGuestbookServletTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.332 sec

Results :

Tests run: 2, Failures: 0, Errors: 0, Skipped: 0

[INFO] 
[INFO] --- maven-war-plugin:2.3:war (default-war) @ guestbook ---
[INFO] Packaging webapp
[INFO] Assembling webapp [guestbook] in [/Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/target/guestbook-1.0-SNAPSHOT]
[INFO] Processing war project
[INFO] Copying webapp webResources [/Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/src/main/webapp/WEB-INF] to [/Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/target/guestbook-1.0-SNAPSHOT]
[INFO] Copying webapp resources [/Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/src/main/webapp]
[INFO] Webapp assembled in [78 msecs]
[INFO] Building war: /Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/target/guestbook-1.0-SNAPSHOT.war
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ guestbook ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ guestbook ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ guestbook ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:testCompile (default-testCompile) @ guestbook ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ guestbook ---
[INFO] Skipping execution of surefire because it has already been run for this configuration
[INFO] 
[INFO] >>> appengine-maven-plugin:1.8.5-maven3.0:devserver (default-cli) @ guestbook >>>
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ guestbook ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ guestbook ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ guestbook ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:testCompile (default-testCompile) @ guestbook ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ guestbook ---
[INFO] Skipping execution of surefire because it has already been run for this configuration
[INFO] 
[INFO] --- maven-war-plugin:2.3:war (default-war) @ guestbook ---
[INFO] Packaging webapp
[INFO] Assembling webapp [guestbook] in [/Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/target/guestbook-1.0-SNAPSHOT]
[INFO] Processing war project
[INFO] Copying webapp webResources [/Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/src/main/webapp/WEB-INF] to [/Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/target/guestbook-1.0-SNAPSHOT]
[INFO] Copying webapp resources [/Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/src/main/webapp]
[INFO] Webapp assembled in [25 msecs]
[INFO] Building war: /Users/bruno/EclipseProjects/myprojects-workspace/appengine-guestbook-java/target/guestbook-1.0-SNAPSHOT.war
[INFO] 
[INFO] <<< appengine-maven-plugin:1.8.5-maven3.0:devserver (default-cli) @ guestbook <<<
[INFO] 
[INFO] --- appengine-maven-plugin:1.8.5-maven3.0:devserver (default-cli) @ guestbook ---
[WARNING] Error injecting: com.google.appengine.devappserver.DevAppServerRunner
java.lang.NoClassDefFoundError: Lorg/sonatype/aether/RepositorySystem;
	at java.lang.Class.getDeclaredFields0(Native Method)
	at java.lang.Class.privateGetDeclaredFields(Class.java:2317)
	at java.lang.Class.getDeclaredFields(Class.java:1762)
	at com.google.inject.spi.InjectionPoint.getInjectionPoints(InjectionPoint.java:661)
	at com.google.inject.spi.InjectionPoint.forInstanceMethodsAndFields(InjectionPoint.java:366)
	at com.google.inject.internal.ConstructorBindingImpl.getInternalDependencies(ConstructorBindingImpl.java:165)
	at com.google.inject.internal.InjectorImpl.getInternalDependencies(InjectorImpl.java:609)
	at com.google.inject.internal.InjectorImpl.cleanup(InjectorImpl.java:565)
	at com.google.inject.internal.InjectorImpl.initializeJitBinding(InjectorImpl.java:551)
	at com.google.inject.internal.InjectorImpl.createJustInTimeBinding(InjectorImpl.java:865)
	at com.google.inject.internal.InjectorImpl.createJustInTimeBindingRecursive(InjectorImpl.java:790)
	at com.google.inject.internal.InjectorImpl.getJustInTimeBinding(InjectorImpl.java:278)
	at com.google.inject.internal.InjectorImpl.getBindingOrThrow(InjectorImpl.java:210)
	at com.google.inject.internal.InjectorImpl.getProviderOrThrow(InjectorImpl.java:986)
	at com.google.inject.internal.InjectorImpl.getProvider(InjectorImpl.java:1019)
	at com.google.inject.internal.InjectorImpl.getProvider(InjectorImpl.java:982)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1032)
	at org.eclipse.sisu.space.AbstractDeferredClass.get(AbstractDeferredClass.java:48)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:86)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision(InternalFactoryToInitializableAdapter.java:55)
	at com.google.inject.internal.ProviderInternalFactory$1.call(ProviderInternalFactory.java:70)
	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:100)
	at org.eclipse.sisu.plexus.PlexusLifecycleManager.onProvision(PlexusLifecycleManager.java:133)
	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:109)
	at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:55)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:68)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.get(InternalFactoryToInitializableAdapter.java:47)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:997)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1047)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:993)
	at com.google.inject.Scopes$1$1.get(Scopes.java:59)
	at org.eclipse.sisu.inject.LazyBeanEntry.getValue(LazyBeanEntry.java:82)
	at org.eclipse.sisu.plexus.LazyPlexusBean.getValue(LazyPlexusBean.java:51)
	at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:260)
	at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:252)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getConfiguredMojo(DefaultMavenPluginManager.java:459)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:97)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:317)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:152)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:555)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:601)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: java.lang.ClassNotFoundException: org.sonatype.aether.RepositorySystem
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:259)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:235)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:227)
	... 57 more
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.974s
[INFO] Finished at: Thu Jan 16 07:09:40 BRT 2014
[INFO] Final Memory: 9M/24M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.appengine:appengine-maven-plugin:1.8.5-maven3.0:devserver (default-cli) on project guestbook: Execution default-cli of goal com.google.appengine:appengine-maven-plugin:1.8.5-maven3.0:devserver failed: A required class was missing while executing com.google.appengine:appengine-maven-plugin:1.8.5-maven3.0:devserver: Lorg/sonatype/aether/RepositorySystem;
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>com.google.appengine:appengine-maven-plugin:1.8.5-maven3.0
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/Users/bruno/.m2/repository/com/google/appengine/appengine-maven-plugin/1.8.5-maven3.0/appengine-maven-plugin-1.8.5-maven3.0.jar
[ERROR] urls[1] = file:/Users/bruno/.m2/repository/org/sonatype/sisu/sisu-inject-bean/2.1.1/sisu-inject-bean-2.1.1.jar
[ERROR] urls[2] = file:/Users/bruno/.m2/repository/org/sonatype/sisu/sisu-guice/2.9.4/sisu-guice-2.9.4-no_aop.jar
[ERROR] urls[3] = file:/Users/bruno/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.14/plexus-interpolation-1.14.jar
[ERROR] urls[4] = file:/Users/bruno/.m2/repository/org/codehaus/plexus/plexus-utils/2.0.6/plexus-utils-2.0.6.jar
[ERROR] urls[5] = file:/Users/bruno/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.5.5/plexus-component-annotations-1.5.5.jar
[ERROR] urls[6] = file:/Users/bruno/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[ERROR] urls[7] = file:/Users/bruno/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[8] = file:/Users/bruno/.m2/repository/org/sonatype/aether/aether-util/1.13.1/aether-util-1.13.1.jar
[ERROR] urls[9] = file:/Users/bruno/.m2/repository/com/google/guava/guava/14.0.1/guava-14.0.1.jar
[ERROR] urls[10] = file:/Users/bruno/.m2/repository/com/google/appengine/appengine-tools-sdk/1.8.4/appengine-tools-sdk-1.8.4.jar
[ERROR] urls[11] = file:/Users/bruno/.m2/repository/com/google/appengine/appengine-local-endpoints/1.8.4/appengine-local-endpoints-1.8.4.jar
[ERROR] urls[12] = file:/Users/bruno/.m2/repository/com/google/appengine/appengine-endpoints/1.8.4/appengine-endpoints-1.8.4.jar
[ERROR] urls[13] = file:/Users/bruno/.m2/repository/com/google/appengine/appengine-api-1.0-sdk/1.8.4/appengine-api-1.0-sdk-1.8.4.jar
[ERROR] urls[14] = file:/Users/bruno/.m2/repository/javax/servlet/servlet-api/2.5/servlet-api-2.5.jar
[ERROR] urls[15] = file:/Users/bruno/.m2/repository/eu/infomas/annotation-detector/3.0.1/annotation-detector-3.0.1.jar
[ERROR] urls[16] = file:/Users/bruno/.m2/repository/org/slf4j/slf4j-nop/1.7.5/slf4j-nop-1.7.5.jar
[ERROR] urls[17] = file:/Users/bruno/.m2/repository/org/slf4j/slf4j-api/1.7.5/slf4j-api-1.7.5.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------: org.sonatype.aether.RepositorySystem
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/AetherClassNotFound

```

Thank you.

--- a/pom.xml
+++ b/pom.xml
@@ -10,9 +10,13 @@
     <groupId>com.google.appengine.demos</groupId>
     <artifactId>guestbook</artifactId>
 
+    <prerequisites>
+        <maven>3.1.0</maven>
+    </prerequisites>
+
     <properties>
         <appengine.app.version>1</appengine.app.version>
-        <appengine.target.version>1.9.20</appengine.target.version>
+        <appengine.target.version>1.9.21</appengine.target.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     </properties>
 
@@ -39,13 +43,13 @@
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
-            <version>4.10</version>
+            <version>4.12</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-all</artifactId>
-            <version>1.9.0</version>
+            <version>1.10.19</version>
             <scope>test</scope>
         </dependency>
         <dependency>
@@ -61,10 +65,10 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
-      
+
     <build>
-        <outputDirectory>target/${project.artifactId}-${project.version}/WEB-INF/classes</outputDirectory> 
-                                   
+        <outputDirectory>target/${project.artifactId}-${project.version}/WEB-INF/classes</outputDirectory>
+
         <plugins>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
@@ -97,6 +101,19 @@
                 <source>1.7</source>
                 <target>1.7</target>
               </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>versions-maven-plugin</artifactId>
+                <version>2.1</version>
+                <executions>
+                    <execution>
+                        <phase>compile</phase>
+                        <goals>
+                            <goal>display-dependency-updates</goal>
+                        </goals>
+                    </execution>
+                </executions>
             </plugin>
         </plugins>
     </build>

